packages/playwright/src/reporters/junit.ts

```diff
@@ -144,65 +144,58 @@ class JUnitReporter implements ReporterV2 {
       attributes: {
         // Skip root, project, file
         name: namePrefix + test.titlePath().slice(3).join(' › '),
         // filename
         classname: suiteName,
+        path: suiteName,
         time: (test.results.reduce((acc, value) => acc + value.duration, 0)) / 1000
-
       },
       children: [] as XMLEntry[]
     };
-    entries.push(entry);
-
-    // Xray Test Management supports testcase level properties, where additional metadata may be provided
-    // some annotations are encoded as value attributes, other as cdata content; this implementation supports
-    // Xray JUnit extensions but it also agnostic, so other tools can also take advantage of this format
-    const properties: XMLEntry = {
-      name: 'properties',
-      children: [] as XMLEntry[]
-    };
-
-    const annotations = [...test.annotations, ...test.results.flatMap(r => r.annotations)];
-    for (const annotation of annotations) {
-      const property: XMLEntry = {
-        name: 'property',
-        attributes: {
-          name: annotation.type,
-          value: (annotation?.description ? annotation.description : '')
-        }
-      };
-      properties.children?.push(property);
-    }
-
-    if (properties.children?.length)
-      entry.children.push(properties);
 
     if (test.outcome() === 'skipped') {
-      entry.children.push({ name: 'skipped' });
+      const skippedEntry = { ...baseTestCaseEntry, children: [{ name: 'skipped' }] };
+      entries.push(skippedEntry);
       return;
     }
 
-    if (!test.ok()) {
-      entry.children.push({
-        name: 'failure',
+    // Collect across all retries
+    const systemOut = [];
+    const systemErr = [];
+
+    // Create a testcase per retry result
+    for (const currentResult of test.results) {
+      const testCaseEntry = {
+        ...baseTestCaseEntry,
         attributes: {
-          message: `${path.basename(test.location.file)}:${test.location.line}:${test.location.column} ${test.title}`,
-          type: 'FAILURE',
+          ...baseTestCaseEntry.attributes,
+          time: currentResult.duration / 1000,
         },
-        text: stripAnsiEscapes(formatFailure(nonTerminalScreen, this.config, test))
-      });
-    }
+      };
 
-    const systemOut: string[] = [];
-    const systemErr: string[] = [];
-    for (const result of test.results) {
-      systemOut.push(...result.stdout.map(item => item.toString()));
-      systemErr.push(...result.stderr.map(item => item.toString()));
-      for (const attachment of result.attachments) {
+      // Add failure per retry failure
+      if (currentResult.error) {
+        testCaseEntry.children.push({
+          name: 'failure',
+          attributes: {
+            message: `${path.basename(test.location.file)}:${test.location.line}:${test.location.column} ${test.title}`,
+            type: 'FAILURE',
+          },
+          text: stripAnsiEscapes(formatFailure(nonTerminalScreen, this.config, test))
+        });
+      }
```

Comment:
the failure shape [stays the same](https://github.com/microsoft/playwright/pull/35323/files#diff-7305b66784f01e1e4d46d104516f9713370aac65ebed328244cc7f2020bdd4c0L191-L196)